thorlcr/activities/xmlwrite/thxmlwrite.cpp

```diff
@@ -66,6 +66,28 @@ class CXmlWriteActivityMaster : public CWriteMasterBase
             supplied.set(helper->getFooter());
             props.setPropInt("@footerLength", buildJsonFooter(s.clear(), supplied, rowTag).length());
         }
+        else
+        {
+            StringBuffer supplied(helper->getHeader());
+            size32_t headerLength = supplied.length();
+            if (headerLength == 0 )
+                headerLength = supplied.set(DEFAULTXMLHEADER).newline().length();
+
+            supplied.set(helper->getFooter());
+            size32_t footerLength = supplied.length();
+            if (footerLength == 0 )
+                footerLength = supplied.set(DEFAULTXMLFOOTER).newline().length();
+
+            IPartDescriptorIterator *parts = fileDesc->getIterator();
+            if (parts->first())
+            {
+                do {
+                    IPartDescriptor &part = parts->query();
+                    part.queryProperties().setPropInt("@headerLength", headerLength);
+                    part.queryProperties().setPropInt("@footerLength", footerLength);
+                } while (parts->next());
```

Comment:
Again, see note above about avoiding per part lengths when not necessary (and when ECL is writing the file it's never necessary).